src/embed.ts

```diff
@@ -19,19 +19,22 @@
  * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
  * @link      https://github.com/serlo-org/serlo.org-cloudflare-worker for the canonical source repository
  */
-import { either as E, option as O } from 'fp-ts'
+import { option as O } from 'fp-ts'
 import * as t from 'io-ts'
 
-import { Url } from './utils'
+import { SentryReporter, Url } from './utils'
 
-export async function embed(request: Request): Promise<Response | null> {
+export async function embed(event: FetchEvent): Promise<Response | null> {
   // example url: embed.serlo.org/thumbnail?url=...
-  const url = Url.fromRequest(request)
+  const url = Url.fromRequest(event.request)
 
   if (url.subdomain !== 'embed') return null
 
+  const sentry = new SentryReporter({ event, service: 'embed' })
```

Comment:
Hm... Shouldn't sentry be global instead? If possible in its own module so all (sub-)functions can import sentry if needed. If not, as an always present argument to all functions called by `handleFetchEvent`?

Reply:
No, the tool we use ( https://www.npmjs.com/package/toucan-js ) needs access to the original `FetchEvent` in order to call `event.waitUntil` (see https://developers.cloudflare.com/workers/learning/fetch-event-lifecycle and https://github.com/robertcepa/toucan-js/blob/master/src/index.ts#L199). Otherwise the execution environment would shut down before the sentry event is send.

So my attempt is that modules receive `event` or `request` (whatever they need) and can create their sentry module in it. In the current way we make sure, that the tag `service` is always set (otherwise `SentryReporter` cannot be initiated). 

Although it is possible to create the sentry object in `handleFetchEvent` and to pass it whenever it is needed.

Reply:
See also https://github.com/robertcepa/toucan-js/issues/48

Reply:
In this case, I'd personally prefer creating the sentry object in `handleFetchEvent` once. But do what you prefer.